uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-action-dropdown.spec.js

```diff
@@ -0,0 +1,1541 @@
+              expect(vm.actions).toEqual(['view']);
+            });
+          });
+        });
+
+        describe('when status is "rejected"', function () {
+          beforeEach(function () {
+            setRequestStatus(request, 'rejected');
+          });
+
+          describe('when not expired', function () {
+            beforeEach(function () {
+              makeRequestExpired(request, false);
+              compileComponent();
+            });
+
+            it('allows correct actions', function () {
+              expect(vm.actions).toEqual(['view']);
+            });
+          });
+
+          describe('when expired', function () {
+            beforeEach(function () {
+              makeRequestExpired(request, true);
+              compileComponent();
+            });
+
+            it('allows correct actions', function () {
+              expect(vm.actions).toEqual(['view']);
+            });
+          });
+        });
+
+        describe('when status is "cancelled"', function () {
+          beforeEach(function () {
+            setRequestStatus(request, 'cancelled');
+          });
+
+          describe('when not expired', function () {
+            beforeEach(function () {
+              makeRequestExpired(request, false);
+              compileComponent();
+            });
+
+            it('allows correct actions', function () {
+              expect(vm.actions).toEqual(['view']);
+            });
+          });
+
+          describe('when expired', function () {
+            beforeEach(function () {
+              makeRequestExpired(request, true);
+              compileComponent();
+            });
+
+            it('allows correct actions', function () {
+              expect(vm.actions).toEqual(['view']);
+            });
+          });
+        });
+      });
+    });
+
+    describe('wants to change status of leave request', function () {
+      // Any action, role or request could be specified here
+      var action = 'approve';
+
+      beforeEach(function () {
+        // Any role or request could be specified here
+        role = 'admin';
+        request = getRequestByType('leave');
+
+        spyOn(request, action).and.returnValue($q.resolve());
+        resolveDialogWith(null);
+        vm.act(action);
+        $rootScope.$digest();
+        compileComponent();
+      });
+
+      it('sees a confirmation dialog', function () {
+        expect(dialog.open).toHaveBeenCalled();
+      });
+
+      describe('confirms the action', function () {
+        beforeEach(function () {
+          resolveDialogWith(true);
+          spyOn($rootScope, '$emit');
+          vm.act(action);
+          $rootScope.$digest();
+        });
+
+        it('other controllers are notified', function () {
+          expect($rootScope.$emit).toHaveBeenCalledWith('LeaveRequest::edit', jasmine.any(Object));
+        });
+      });
+    });
+
+    describe('wants to delete leave request', function () {
```

Comment:
This is basically the same test as before right? The only thing (as far as i can see) it's really the different even name. I think you can simply have one single test for a single action, not point having a single test just for a different event name